### PR TITLE
Add e2e test for automatic sidecar injection

### PR DESCRIPTION
**Pull request by ostromart**
Reviving https://github.com/istio/istio/pull/4926.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -51,6 +51,7 @@ install/kubernetes/istio-one-namespace.yaml
 install/kubernetes/istio-sidecar-injector-configmap-debug.yaml
 install/kubernetes/istio-sidecar-injector-configmap-release.yaml
 install/kubernetes/istio-sidecar-injector.yaml
+install/kubernetes/istio-auth-sidecar-injector.yaml
```
Comment (costinm) [inline .gitignore]: Please stop adding more files here - we need to get rid of the existing ones instead.

Run helm template with the right parameters, generate a file in the $OUT directory - and change the test to read from there.

No 'updateVersion.sh' either - just helm template (in a make target)

File: install/updateVersion.sh [hunk 2]
```diff
@@ -59,7 +59,9 @@ if [[ ! -z ${ISTIO_RELEASE:-} ]]; then
         gen_file $target ${DEST_DIR}
     done
 else
-    for target in istio.yaml istio-auth.yaml istio-one-namespace.yaml istio-one-namespace-auth.yaml istio-multicluster.yaml istio-auth-multicluster.yaml istio-remote.yaml istio-galley.yaml istio-auth-galley.yaml;do
+    for target in istio.yaml istio-auth.yaml istio-one-namespace.yaml istio-one-namespace-auth.yaml \
```
Comment (costinm) [inline install/updateVersion.sh]: See above :-)

Extra points of you remove updateVersion.sh

File: prow/e2e-cluster_wide-auth.sh [hunk 3]
```diff
@@ -31,4 +31,4 @@ set -x
 ROOT=$(cd $(dirname $0)/..; pwd)
 
 echo 'Running cluster-wide e2e rbac, auth Tests'
-${ROOT}/prow/e2e-suite.sh --test_vm --auth_enable --cluster_wide --use_galley_config_validator "$@"
+${ROOT}/prow/e2e-suite.sh --test_vm --auth_enable --cluster_wide --use-automatic-injection --use_galley_config_validator "$@"
```
Comment (costinm) [inline prow/e2e-cluster_wide-auth.sh]: Shouldn't we have separate suites, one with automatic one with manual ? Or do we already have ?

File: tests/e2e/framework/kubernetes.go [hunk 4]
```diff
@@ -147,21 +147,27 @@ type KubeInfo struct {
 
 func getClusterWideInstallFile() string {
 	const (
-		nonAuthInstallFile           = "istio.yaml"
-		authInstallFile              = "istio-auth.yaml"
-		nonAuthWithGalleyInstallFile = "istio-galley.yaml"
-		authWithGalleyInstallFile    = "istio-auth-galley.yaml"
+		nonAuthInstallFile                    = "istio.yaml"
```
Comment (costinm) [inline tests/e2e/framework/kubernetes.go]: How about passing this as a param or env variable ?

File: tests/e2e/framework/kubernetes.go [hunk 5]
```diff
@@ -147,21 +147,27 @@ type KubeInfo struct {
...
+		nonAuthWithGalleyInstallFile          = "istio-galley.yaml"
+		authWithGalleyInstallFile             = "istio-auth-galley.yaml"
+		nonAuthWithSidecarInjectorInstallFile = "istio-sidecar-injector.yaml"
+		authWithSidecarInjectorInstallFile    = "istio-auth-sidecar-injector.yaml"
 	)
 	var istioYaml string
 	if *authEnable {
 		if *useGalleyConfigValidator {
 			istioYaml = authWithGalleyInstallFile
+		} else if *useAutomaticInjection {
```
Comment (ayj) [inline tests/e2e/framework/kubernetes.go]: This implies that galley and sidecarInjector are mutually exclusive but https://github.com/istio/istio/pull/6124/files#diff-d11449323523da124c3676e8b48e4597R34 enables both tests. I think we do _want_ to test both features at the same time in cluster wide tests, but we might need to refactor the helm setup files accordingly. Galley is now enabled by default in istio.yaml. We might be able to make a similar change in the e2e tests when clusterWide=true

**Pull request closed without merging**